### ISSUE TITLE
Better newlines and long texts

### ISSUE THREAD

**Issue by tjconcept** [citycide/tablemark#4]
The current method of converting to strings does not take newlines into consideration, which are probably not that uncommon. The current state of "newlines in markdown tables" seems pretty bad from a quick search on the internet.

Current:

```sh
echo '{"message":"Wr\nong"}' | tablemark
| Message |
| ------- |
| Wr
ong  |
```

### Replacing `\n` with `<br />`:

```
| Message     | Code  |
| ----------- | ----- |
| Wr<br />ong | 2     |
```

| Message     | Code  |
| ----------- | ----- |
| Wr<br />ong | 2     |

#### Keeping gutters/separators

```markdown
| Message | Code  |
| ------- | ----- |
| Wr      | 2     |
| ong     |       |
| Good    | 0     |
```

| Message | Code  |
| ------- | ----- |
| Wr      | 2     |
| ong     |       |
| Good    | 0     |

#### Without gutters/separators

```
| Message | Code  |
| ------- | ----- |
| Wr      | 2     |
  ong
| Good    | 0     |
```

| Message | Code  |
| ------- | ----- |
| Wr      | 2     |
  ong
| Good    | 0     |

Since only the first produces valid markdown (GitHub-usable), I think that should definitely be an option. Apart from that consideration, I'm mostly into the last one (wrapping and aligning but without gutters/separators).

My rough proposal is:
Wrap at newlines and after any 80 characters run. Do not add gutters and separators, but align content. Provide an option to tweak characters per line or disable automatic line wrapping. Provide an option to replace newlines with `<br />` and disable any wrapping.

**Comment by citycide** [comment]
Yo @tjconcept - thanks for the writeup. I'm back from vacation ( a terrible non-vacation really since it resulted in a totaled car and leaving early due to hurricane ) and will review this and your other PRs when I get a chance.

Right now I think your proposal is the way to go, although defaulting to `<br />` might be a choice.

**Comment by tjconcept** [comment]
I'm sorry to hear about your vacation, but welcome back.

I went ahead and implemented this for the challenge, first as a prototype, then I couldn't stop. It was.. *much* more complex than I had imagined. Things as word wrapping (visual length ≠ `.length` due to unicode), and generating and rendering a matrix for each row.
Anyway, I am satisfied with the result. I'm gonna put up a PR for you to peak at the code soon.

> though maybe make `<br />` the default.

I was inclined to newlines because my primary use case is "pretty printing" in the terminal. But I guess this package (reasonably) has "valid markdown" as the primary use case.

**Comment by citycide** [comment]
@tjconcept Either way shouldn't be too bad as a default. A shorthand flag on the CLI would make either one easy enough to trigger, like `-n` for using newlines or `-b` for breaks or something.

Unicode lengths may need a little extra love. Might be worth adding in [string-length](https://www.npmjs.com/package/string-length) for that but there could also be alternatives. 

I've made you a collaborator for both the module & the CLI so you can make pushes. I'll just jump in with reviews as we've been doing, but you've been putting in a load of time for these so it'll be easier for you to work on things. Thanks for your efforts 👍

**Comment by tjconcept** [comment]
> Unicode lengths may need a little extra love. Might be worth adding in string-length for that but there could also be alternatives.

I ended up writing a new package for this as we need the wrapped result back as an array, not a string, we need the string length counter to be tied to the wrapper (returning the same result), and we need *hard wrapping* (some packages had hard wrapping, but would still overflow on words longer than the column).
It's not *that* bad to count too much, as long as you wrap by the same the same metric, and only wraps on explicit white space. It will only result in some excess whitespace.

> I've made you a collaborator for both the module & the CLI so you can make pushes. I'll just jump in with reviews as we've been doing, but you've been putting in a load of time for these so it'll be easier for you to work on things.

Awesome! Give PRs an approval when you think they are ready then.

> Thanks for your efforts 👍

No problem. I'm scratching my own itches here (this being the last one I think, for now).

**Comment by citycide** [comment]
Closed by #6
